synchronizer/l1_live_block_ranges.go

```diff
@@ -5,23 +5,26 @@ import (
 	"fmt"
 )
 
-type liveBlockRangeItem struct {
+type liveBlockRangeItem[T any] struct {
 	blockRange blockRange
+	tag        T
 }
 
-type liveBlockRanges struct {
-	ranges []liveBlockRangeItem
+type liveBlockRangesGeneric[T any] struct {
```

Comment:
You love to use everything generics

Reply:
Yes!!! here you take advantge of this object to be reused